### ISSUE TITLE
Introduce Lazy Loading to the App

### ISSUE THREAD

**Issue by Moejoe90** [Neufund/platform-frontend#2670]
**Background**
As per frontend meeting during the retreat
The usage of React lazy introduces a major bundle size reshift as components and libraries will be loaded per request. 

I suggest implementing this ticket on a couple of steps

- [x] Introduce React Lazy to Major components `Dashboard` `Settings` etc  
During this step, we should analyze how much size are we saving. 

- [ ] Lazy load smaller components and resources 

- [ ] Lazy load libraries based on wallet type etc 

- [ ] Lazy load modules (sagas/reducers/selectors) on demand

**Designs**
N/A
**Acceptance criteria**
Everything should work as normal. 
**Technical notes**
The bundle size should decrease
**Testing notes**
cannot be tested by QA

**Comment by desfero** [comment]
@Moejoe90 the first point was already done here https://github.com/Neufund/platform-frontend/pull/2574. But is blocked by this issue https://github.com/airbnb/enzyme/pull/1975. I'm watching for new enzyme release and when it's done will reopen pr mentioned above.
